Marlin/src/HAL/shared/Delay.h

```diff
@@ -157,5 +144,10 @@
 // Delay in nanoseconds
 #define DELAY_NS(x) DELAY_CYCLES((x) * ((F_CPU) / 1000000UL) / 1000UL)
 
-// Delay in microseconds
-#define DELAY_US(x) DELAY_CYCLES((x) * ((F_CPU) / 1000000UL))
+#ifdef IS_TEENSY_40_41
```

Comment:
Here it would be better to check the GCC version instead. We have `GCC_VERSION` defined so we can use…
```cpp
#if GCC_VERSION < 40602 // Test for GCC < 4.6.2
  // . . .
#endif
```

Reply:
Checking the GCC version should be done around the `DELAY_CYCLES` macro declaration. The issue with this, is that I don't have a working workaround for GCC < 6.3 (my code is not accepted by such old compiler). So, instead, I'm replacing the call to DELAY_US for Teensy, because, on this platform, it's the only macro that's being used by both constant and non constant expression (said differently, on Teensy, both DELAY_NS and DELAY_CYCLES are always called with compile time parameters). Hopefully, our precision on microsecond delay is good enough with the DWT loop that detecting compile-time constant is useless for this unit case anyway. 

Another possible implementation would be to make per-platform DELAY_US macro declaration so that all ARM platform skip the compile time detection for µs delay anyway. Probably cleaner and that would remove the ugly `if teensy` check.